File: java_sim/simulation_188623/src/Owca.java
import javax.swing.ImageIcon;

public class Owca extends Zwierze {

    Owca(Swiat swiatOrganizmu, int x_pos, int y_pos, int wiek, boolean czyCzeka) {
        super(swiatOrganizmu, 4, 4, x_pos, y_pos, wiek, czyCzeka);
        this.ikonka = new ImageIcon("zdjecia/owca.png");
        this.hexIkonka = new ImageIcon("zdjecia/owcaHex.png");
    }
    
    @Override
    String getNazwa() {        
        return ("Owca");
    }

    @Override
    char getZnaczek() {
        return '7'; 
    }

    @Override
    Organizm klonowanie(int posX, int posY) {        
        return new Owca(swiatOrganizmu, posX, posY, 1, true);
    }
}

File: cpp_sim/188623/Owca.cpp
#include "Owca.h"

Owca::Owca(Swiat* swiatOrganizmu, int posX, int posY, int age, bool czyCzeka)
: Zwierze(swiatOrganizmu, 4, 4, posX, posY, age, czyCzeka){
    
};

void Owca::rysowanie(){
    printf("O");
};

string Owca::getNazwa(){
    return ("Owca");
};

Organizm *Owca::klonowanie(int posX, int posY){
    return new Owca(swiatOrganizmu, posX, posY, 1, true);
};

Owca::~Owca(){
    
};
